pos_event_calendar/static/src/js/custom_button.js

```diff
@@ -0,0 +1,31 @@
+import {Component} from "@odoo/owl";
+import {usePos} from "@point_of_sale/app/store/pos_hook";
+import {useService} from "@web/core/utils/hooks";
+import {ControlButtons} from "@point_of_sale/app/screens/product_screen/control_buttons/control_buttons";
+import {patch} from "@web/core/utils/patch";
+import {EventSelectorPopup} from "./Popups/EventSelectorPopup/EventSelectorPopup";
+
+export class EventButton extends Component {
+    static template = "pos_event_calendar.EventButton";
+   
+    setup() {
+        super.setup();
+        this.pos = usePos();
+        this.ui = useService("ui");
+        this.dialog = services.dialog;
+    }
+
+}
+
+patch(ControlButtons.prototype, {
+    components: {
+        ...ControlButtons.components,
+        EventButton,
+    },
+    /**
+     * Called when the button is clicked. Open the EventSelectorPopup.
+     */
+    async onClick() {
```

Comment:
`onClick` is too generic and it may collide with other modules.
Please use a more concise name: e.g.: 

```suggestion
    async onClickEventSelector() {
```